x/campaign/types/mainnet_vesting_account.go

```diff
@@ -1,11 +1,12 @@
 package types
 
+import "errors"
+
 // GetTotalShares return total shares for account and delayed vesting options
 func (m MainnetVestingAccount) GetTotalShares() (Shares, error) {
-	vestingShares, err := m.VestingOptions.GetDelayedVestingShare()
-	if err != nil {
-		return nil, err
+	dv := m.VestingOptions.GetDelayedVesting()
+	if dv == nil {
+		return nil, errors.New("invalid vesting options type")
```

Comment:
```suggestion
		return nil, ErrInvalidVestingOption
```

maybe should we create the `ErrInvalidVestingOption` error?

Reply:
I think it's okay. Methods in `types` are last-layer called methods so they are usually wrapped inside a higher-level error.
We generally define errors for `keeper` logic